quickwit-cluster/src/cluster.rs

```diff
@@ -339,21 +357,26 @@ impl Cluster {
 
 fn parse_available_services_val(
     available_services_val: &str,
-    member: &Member,
-) -> Vec<QuickwitService> {
-    let mut available_services = vec![];
+    node_unique_id_str: &str,
+) -> anyhow::Result<HashSet<QuickwitService>> {
+    let mut available_services = HashSet::new();
+    // Case where a node was started without any services.
+    if available_services_val.is_empty() {
+        return Ok(available_services);
+    }
     let available_services_str = available_services_val.split(',');
     for service_str in available_services_str {
-        match QuickwitService::try_from(service_str) {
-            Ok(service) => available_services.push(service),
-            Err(_) => error!(
+        if let Ok(service) = QuickwitService::try_from(service_str) {
+            available_services.insert(service);
+        } else {
+            anyhow::bail!(
```

Comment:
There is a valid case for not bailing and ignoring the member with a warning when an operator performs a rolling upgrade of a cluster and the upgrade introduces a new service.

Reply:
well spotted